### DISCUSSION TITLE
Build increased by 100% when set "NG_BUILD_MANGLE=false"

### DISCUSSION THREAD

**Discussion by rbernalagainagain** [archimedes-projects/archimedes-js#12] [Q&A]
In the Angular projects, the size of the build in increased more than 100% when set in the application NG_BUILD_MANGLE=false.

example: **NG_BUILD_MANGLE=false.**

> <img width="446" alt="Captura de pantalla 2021-10-17 a las 14 48 04" src="https://user-images.githubusercontent.com/37298352/137627836-f68c8510-d392-48f4-93a4-fbb4f771bded.png">

example: **NG_BUILD_MANGLE=true**

> <img width="487" alt="Captura de pantalla 2021-10-17 a las 14 50 46" src="https://user-images.githubusercontent.com/37298352/137627919-011ac604-b6e9-447a-acd9-0d2b822009f4.png">

I don't consider this an issue, but I don't think it's much of a sacrifice for the applications.
Angular is very restrictive with its core, and I can't find a possible solution quickly.

What possible solution can we find?

**Comment by cesalberca** [comment]
It's a pity that Angular doesn't optimize the code correctly with that option set. We use the name of the classes to handle caching so we can't have their name minificated. A workaround might be to add a way to provide custom cache keys in order to be able to optimize the code. For now I'll create an issue on Angular's repo and see if it can be fixed.

**Reply by esuca** [reply]
What if we change the implementation of our cache? I could be wrong, but maybe within the cache decorator we could define a new metadata key using reflection for each command or query.
Another workaround could be to use instanceof, but I'm not sure.

**Reply by cesalberca** [reply]
So maybe instead of extending from Query or Command we could have a decorator? The problem with that is that without extending from an abstract class forces you to implement the right method. And doing both is a little bit redundant, no?

**Reply by esuca** [reply]
If we have a decorator we need an interface to force the user to use the right method and have better typescript type inference. Also with a decorator we don't need to have the `internalExecute()` method... Take a look at [Nestjs CQRS implementation](https://github.com/nestjs/cqrs). For example the [@QueryHandler ](https://github.com/nestjs/cqrs/blob/master/src/decorators/query-handler.decorator.ts) only defines the metadata, then inside the [QueryBus](https://github.com/nestjs/cqrs/blob/master/src/query-bus.ts) the handler is registered using the metadata defined before.
I'm sure we could do something similar in Archimedes
